### PR TITLE
Feature/simplified build rebased

### PR DESCRIPTION
**Pull request by florianesser-tng**
Provide github build action for reliable and reproducable builds

### CODE REVIEW

File: .github/workflows/build-installer.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          node-version: '20'
+
+      - name: setup npm project
+        working-directory: "WebUI"
+        run: npm install
+
+      - name: build installer
+        working-directory: "WebUI"
+        run: |
+          npm install
```
Comment (marijnvg-tng) [inline .github/workflows/build-installer.yml:56]: Is this second `npm install` necessary?

Comment (florianesser-tng) [reply .github/workflows/build-installer.yml:56]: no, thanks for spotting

File: WebUI/build/README.md [hunk 2]
```diff
@@ -1,26 +1,18 @@
...
+   - Create a reference conda environment with libuv installed
+    ```
+    conda create -n cp311_libuv python=3.11 libuv -y
+    # copy the path to this conda env
+    conda env list | findstr cp311_libuv
+    ```
+    - symlink or copy the conda env path into /python_package_resources
+3. run ```npm run pack-python```
+
+The resulting `WebUI/npm_package_res/env.7z` could be reused for all platforms.
```
Comment (marijnvg-tng) [inline WebUI/build/README.md:16]: In what circumstance does this need to be executed and reference to GitHub action

File: WebUI/build/scripts/pack-offline.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+function main() {
+    const workDir = path.join(__dirname, `env-${platform}`);
+
+    const offlineEnvDir = unzippedPackagedPyenv(zippedPyenv)
+    const pythonExe = existingFileOrExit(path.join(offlineEnvDir, 'python.exe'));
+
+    const platformSpecificRequirementsTxt = existingFileOrExit(path.join(__dirname, '..', '..', 'service', `requirements-${platform}.txt`));
+    const requirementsTxt = existingFileOrExit(path.join(__dirname, '..', '..', 'service', `requirements-${platform}.txt`));
+    runPipInstall(platformSpecificRequirementsTxt, offlineEnvDir)
+    runPipInstall(requirementsTxt, offlineEnvDir)
```
Comment (marijnvg-tng) [inline WebUI/build/scripts/pack-offline.js:87]: Two identical tasks? Seperations between offline/online

**Review by marijnvg-tng — CHANGES_REQUESTED** [review]
see the comments...

File: WebUI/build/README.md (hunk 2 above)
Comment (florianesser-tng) [reply WebUI/build/README.md:16]: I am not sure, if I understand the question. Do you mean, what we do with the env.7z?

This packaged empty is "as is" newly setup python env. It awaits the correct pip install commands to run to enrich for the platform specific) dependencies so the app can bootup the backend. As there are no dependencies installed, this env.7z can be used for any of the builds : arc/ultra/ultra2 and does not need to be rebuilt in general.

The npm build scripts expect this to be present, and so does the github action (at the correct point in time). The build scripts will fail, if it is not present

File: WebUI/build/scripts/pack-offline.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    pipInstall.stderr.on('data', (data) => {
+        console.error(data.toString());
+    });
+    pipInstall.on('close', (code) => {
+        if (code !== 0) {
+            console.error('Failed to install requirements');
+            process.exit(1);
+        }
+    });
+}
```
[the remark this answers is not in the document]
Comment (florianesser-tng) [reply WebUI/build/scripts/pack-offline.js:79]: added step to execute get-pip.py

File: WebUI/build/scripts/pack-offline.js (hunk 3 above)
Comment (florianesser-tng) [reply WebUI/build/scripts/pack-offline.js:106]: loaded base requirements.txt for the second runPipInstall

**Pull request merged**